### PR TITLE
fix(react-accordion): deprecate navigation prop

### PR DESCRIPTION
**Pull request by mainframev**
As reported in #30799 , deprecated navigation prop in order to align react-accordion behaviour in the next version with [w3 APG pattern](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/examples/accordion/)

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr> | `100.008 kB`<br />`30.527 kB` | `83.428 kB`<br />`26.256 kB` | `-16.58 kB` <br />`-4.271 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `212.636 kB`<br />`61.23 kB` | `212.573 kB`<br />`61.981 kB` | `-63 B` <br />`751 B` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.092 MB`<br />`269.637 kB` | `1.092 MB`<br />`269.61 kB` | `-58 B` <br />`-27 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.141 kB`<br />`20.157 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.442 kB`<br />`14.607 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/48ff23e70fe47f8f91dcbaceb0cb7f3621bca339'>48ff23e70fe47f8f91dcbaceb0cb7f3621bca339</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">33</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">40</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/Avatar-mount_base.html">612</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/Avatar-mount.html">649</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/Button-mount_base.html">300</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/Button-mount.html">306</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/Field-mount_base.html">1128</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/Field-mount.html">1146</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/FluentProvider-mount_base.html">704</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/FluentProvider-mount.html">711</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/FluentProviderWithTheme-mount_base.html">87</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/FluentProviderWithTheme-mount.html">83</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">33</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">40</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">79</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/MakeStyles-mount_base.html">862</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/MakeStyles-mount.html">864</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/Persona-mount_base.html">1763</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/Persona-mount.html">1698</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/SpinButton-mount_base.html">1388</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/SpinButton-mount.html">1430</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/SwatchPicker-mount_base.html">1541</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31587/perf-test-react-components/SwatchPicker-mount.html">1564</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-accordion-c3e0ad16-1578-49d3-90c0-422d2c2931e0.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-accordion-c3e0ad16-1578-49d3-90c0-422d2c2931e0.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31587&commitId=c8385fe0f1e1c438c7ca471b86bf1f2269f57517&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask.normal.chromium.png|7| Changed|
|Avatar Converged.badgeMask - RTL.normal.chromium.png|0| Added|

</details>


  <details><summary><b>Card Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Card Converged.appearance - RTL.normal.chromium.png|0| Added|

</details>


  <details><summary><b>Card Converged - Interactive</b> 12 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Card Converged - Interactive.appearance interactive - Filled - High Contrast.click.chromium.png|0| Added|
|Card Converged - Interactive.appearance interactive - Filled - High Contrast.hover.chromium.png|0| Added|
|Card Converged - Interactive.appearance interactive - Filled - High Contrast.normal.chromium.png|0| Added|
|Card Converged - Interactive.appearance interactive - Filled - Dark Mode.click.chromium.png|0| Removed|
|Card Converged - Interactive.appearance interactive - Filled - Dark Mode.hover.chromium.png|0| Removed|
|Card Converged - Interactive.appearance interactive - Filled - Dark Mode.normal.chromium.png|0| Removed|
|Card Converged - Interactive.appearance interactive - Filled Alternative - High Contrast.click.chromium.png|0| Removed|
|Card Converged - Interactive.appearance interactive - Filled Alternative - High Contrast.hover.chromium.png|0| Removed|
|Card Converged - Interactive.appearance interactive - Filled Alternative - High Contrast.normal.chromium.png|0| Removed|
|Card Converged - Interactive.appearance interactive - Filled Alternative.click.chromium.png|0| Removed|
|Card Converged - Interactive.appearance interactive - Filled Alternative.hover.chromium.png|0| Removed|
|Card Converged - Interactive.appearance interactive - Filled Alternative.normal.chromium.png|0| Removed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-accordion/src/components/Accordion/Accordion.types.ts [hunk 2]
```diff
@@ -42,6 +42,7 @@ export type AccordionProps<Value = AccordionItemValue> = ComponentProps<Accordio
   multiple?: boolean;
 
   /**
+   * @deprecated Arrow keyboard navigation will be dropped.
```
Comment (smhigley) [inline packages/react-components/react-accordion/src/components/Accordion/Accordion.types.ts:45]:
```suggestion
   * @deprecated Arrow keyboard navigation is not recommended for accordions. Consider using Tree if arrow navigation is a hard requirement.
```
just adding some more specific wording & recommendation

**Review by smhigley — COMMENTED** [review]
One small comment wording suggestion, & it would probably be good to remove the Navigation example from the storybook (the `AccordionNavigation.stories.tsx` file)

I think @bsunderhus should be the one to sign off on the overall deprecation approach

**Comment by Hotell** [comment]
pls make sure to update this branch with latest master - accordion was migrated to new structure. that's why there is no codeowner assigned.

File: packages/react-components/react-accordion/src/components/Accordion/useAccordion.ts [hunk 3]
```diff
@@ -28,6 +29,7 @@ export const useAccordion_unstable = <Value = AccordionItemValue>(
     initialState: [],
   });
 
+  /** @deprecated will be removed after navigation prop is removed */
```
Comment (bsunderhus) [inline packages/react-components/react-accordion/src/components/Accordion/useAccordion.ts:32]: nit: 

```suggestion
  /** FIXME: deprecated will be removed after navigation prop is removed */
```

**Review by bsunderhus — CHANGES_REQUESTED** [review]
Thanks for the contrib @mainframev, looks good to me. I'd follow up on @smhigley  recommendation and remove the navigation story entirely (https://react.fluentui.dev/?path=/docs/components-accordion--default#navigation)

File: packages/react-components/react-accordion/src/components/Accordion/Accordion.types.ts (hunk 2 above)
Comment (mainframev) [reply packages/react-components/react-accordion/src/components/Accordion/Accordion.types.ts]: Great, thanks 👍🏻

File: packages/react-components/react-accordion/src/components/Accordion/useAccordion.ts (hunk 3 above)
Comment (mainframev) [reply packages/react-components/react-accordion/src/components/Accordion/useAccordion.ts]: Thanks! :)

**Review by ValentinaKozlova — APPROVED** [review]

**Review by bsunderhus — APPROVED** [review]

**Comment by smhigley** [comment]
@mainframev it looks ready to merge! Thanks for making this change 🎉

**Pull request merged**